src/c++/perf_analyzer/genai-perf/genai_perf/graphs/plot_manager.py

```diff
@@ -0,0 +1,98 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from genai_perf.graphs.box_plot import BoxPlot
+from genai_perf.graphs.heat_map import HeatMap
+from genai_perf.graphs.scatter_plot import ScatterPlot
+from genai_perf.llm_metrics import Statistics
+
+
+class PlotManager:
+    """
+    Manage details around plots generated
+    """
+
+    def __init__(self, stats: Statistics) -> None:
+        self._stats = stats
+
+    def create_default_graphs(self):
+        bp = BoxPlot(self._stats)
+        bp.create_box_plot(
+            y_key="time_to_first_tokens",
+            graph_title="Time to First Token Analysis",
+            filename_root="ttft",
+            x_label="Time to First Token of Individual Requests (seconds)",
+        )
+        bp.create_box_plot(
+            y_key="request_latencies",
+            graph_title="Total Output Time Analysis",
+            filename_root="tot",
+            x_label="Time to Completion of Individual Requests (seconds)",
+        )
+
+        hm = HeatMap(self._stats)
+        hm.create_heat_map(
+            x_key="num_input_tokens",
+            y_key="num_output_tokens",
+            x_metric="input_tokens",
+            y_metric="generated_tokens",
+            graph_title="Distribution of Input Tokens to Generated Tokens (Over All Requests)",
+            x_label="Number of Input Tokens Per Request",
+            y_label="Number of Generated Tokens Per Request",
+            filename_root="heatmap",
+        )
+
```

Comment:
General comment about filenames: do we want to prepend genai_perf_ before them or something similar? That would make it easy for users to find all of the files. I'm not sure if this is pushing all of them into a directory anyway then, which would make that naming less necessary.

Also, do we want to append the model name? What happens if a user tries to get these plots for two models in a row?

Reply:
I was hoping that gathering all of the artifacts in a specific directory would help with finding all of the output.

Can we have a different discussion about file names including the model name? We have leaned into the user being responsible for not overwriting their data.

Reply:
Got it, I was unclear on whether the code did that. A directory works.

Fair enough regarding naming, it sounds like that's what users would be used to then.